sechub-wrapper-xray/src/main/java/com/mercedesbenz/sechub/wrapper/xray/report/XrayWrapperReportTransformer.java

```diff
@@ -0,0 +1,226 @@
+                // want to add it, cycloneDX only allows integer, but cwes are optional
+            }
+        }
+        vulnerability.setCwes(cwes);
+    }
+
+    private void extractSourceInformation(JsonNode node, Vulnerability vulnerability) {
+        String name = node.get("provider").asText();
+        String url = "";
+        if (name != null) {
+            addSourceToVulnerability(url, name, vulnerability);
+        }
+    }
+
+    private void addSourceToVulnerability(String sourceUrl, String sourceName, Vulnerability vulnerability) {
+        Vulnerability.Source source = new Vulnerability.Source();
+        source.setName(sourceName);
+        source.setUrl(sourceUrl);
+        vulnerability.setSource(source);
+    }
+
+    private void extractRatingFromString(JsonNode node, Vulnerability vulnerability, String cvssString) {
+        // cvssString from security report contains multiple information fragments
+        // (score, method and vector)
+        // example: "cvss_v3": "9.8/CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H"
+        String[] cvssArray = cvssString.split("/", 3);
+
+        String score = cvssArray[0];
+        double scoreDouble = parseDouble(score);
+        String method = cvssArray[1];
+        String vector = cvssArray[2];
+        method = method.replace(":", "v");
+        method = method.replace(".", "");
+        method = method.replace("0", "");
+        String severity = node.get("severity").asText().toLowerCase();
+        String severitySource = node.get("severity_source").asText().toUpperCase();
+
+        if (severity != null && severitySource != null && method != null && vector != null && score != null) {
+            RatingRecord ratingRecord = new RatingRecord(scoreDouble, severity, method, vector, severitySource);
+            addRatingToVulnerability(ratingRecord, vulnerability);
+        }
+    }
+
+    private void addRatingToVulnerability(RatingRecord ratingRecord, Vulnerability vulnerability) {
+        Vulnerability.Rating rating = new Vulnerability.Rating();
+
+        String sourceUrl = "";
+        String id = vulnerability.getId();
+        if (!id.contains("XRAY")) {
+            sourceUrl = "https://nvd.nist.gov/vuln/detail/" + id;
+        }
+
+        Vulnerability.Source source = new Vulnerability.Source();
+        source.setName(ratingRecord.severitySource());
+        source.setUrl(sourceUrl);
+        rating.setSource(source);
+
+        rating.setScore(ratingRecord.scoreDouble());
+
+        rating.setSeverity(Vulnerability.Rating.Severity.fromString(ratingRecord.severity()));
+
+        rating.setMethod(Vulnerability.Rating.Method.fromString(ratingRecord.method()));
+
+        rating.setVector(ratingRecord.vector());
+
+        vulnerability.addRating(rating);
+    }
+
+    private void addDescriptionToVulnerability(JsonNode node, Vulnerability vulnerability) {
+        String description = node.get("component_versions").get("more_details").get("description").asText();
```

Comment:
Strings like "component_versions" etc. should IMO be constants and the constants could be inside `XrayWrapperReportConstants`. Reason - some parts are used multiple times and this would reduce typo possiblity.

But its only a suggestion and not really necessary.